### PR TITLE
LPS-60598

### PR DESCRIPTION
**Pull request by hhuijser**

### CODE REVIEW

File: portal-web/docroot/html/taglib/ui/input_asset_links/page.jsp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 			cssClass="asset-selector"
 			data='<%= (Map<String, Object>)selectorEntry.get("data") %>'
 			id='<%= (String)selectorEntry.get("id") %>'
-			message='<%= (String)selectorEntry.get("message") %>'
+			message='<%= HtmlUtil.escape((String)selectorEntry.get("message")) %>'
```
Comment (samuelkong) [inline portal-web/docroot/html/taglib/ui/input_asset_links/page.jsp]: This causes a double escape in /portal-web/docroot/html/taglib/ui/icon/page.jsp:94
1. Do the escaping in /portal-web/docroot/html/taglib/ui/icon/page.jsp
2. Do a complete review of every use case of <liferay-ui:icon> and if there's an escape on the `message` attribute, you'll need to remove it.


@daviddotzhang

File: modules/apps/asset/asset-browser-web/src/main/resources/META-INF/resources/toolbar.jsp [hunk 2]
```diff
@@ -47,7 +47,7 @@ AssetRendererFactory<?> assetRendererFactory = AssetRendererFactoryRegistryUtil.
 			addPortletURLString = HttpUtil.addParameter(addPortletURLString, "refererPlid", plid);
 			%>
 
-			<aui:nav-item href="<%= addPortletURLString %>" label='<%= LanguageUtil.format(request, "add-x", assetRendererFactory.getTypeName(locale, subtypeSelectionId)) %>' />
+			<aui:nav-item href="<%= addPortletURLString %>" label='<%= LanguageUtil.format(request, "add-x", HtmlUtil.escape(assetRendererFactory.getTypeName(locale, subtypeSelectionId))) %>' />
```
Comment (samuelkong) [inline modules/apps/asset/asset-browser-web/src/main/resources/META-INF/resources/toolbar.jsp]: portal-web/docroot/html/taglib/aui/nav_item/end.jsp

1. Remove the call to `stripHtml` on line 31 and replace the use of `title` on line 39 with `escapeAttribute`
2. Use `escape` on the label on line 55
3. Do a review of all use cases of <aui:nav-item>, remove any escape on the 'label' attribute

**Comment by samuelkong** [comment]
@daviddotzhang @hhuijser 

See comments.

**Comment by daviddotzhang** [comment]
@samuelkong 

thanks for the detailed comments, will continue to work on it.

**Pull request closed without merging**